great_expectations/validation_operators/types/validation_operator_result.py

```diff
@@ -0,0 +1,258 @@
+from typing import Dict, List, Union
+
+from great_expectations.core import ExpectationSuiteValidationResult
+from great_expectations.core.id_dict import BatchKwargs
+from great_expectations.data_context.types.resource_identifiers import (
+    ValidationResultIdentifier,
+)
+from great_expectations.types import DictDot
+
+
+class ValidationOperatorResult(DictDot):
+    """
+    The run_results property forms the backbone of this type and defines the basic contract for what a validation
+    operator's run method returns. It is a dictionary where the top-level keys are the ValidationResultIdentifiers of
+    the validation results generated in the run. Each value is a dictionary having at minimum, validation_result and
+    actions_results keys; this dictionary can contain other keys that are relevant for a specific validation operator
+    implementation. For example, the dictionary from a WarningAndFailureExpectationSuitesValidationOperator
+    would have an extra key named "expectation_suite_severity_level" to indicate if the suite is at either a
+    "warning" or "failure" level.
+
+    e.g.
+    {
+        ValidationResultIdentifier: {
+            "validation_result": ExpectationSuiteValidationResult,
+            "actions_results": {}
+        }
+    }
+
+    Most other properties of this class are derived from the run_results property using lazy evaluation.
+    """
+
+    def __init__(
+        self,
+        run_id: str,
+        run_results: Dict[
+            ValidationResultIdentifier,
+            Dict[str, Union[ExpectationSuiteValidationResult, dict, str]],
+        ],
+        validation_operator_config,
+        evaluation_parameters: dict = None,
+        success: bool = None,
+    ) -> None:
+        self._run_id = run_id
```

Comment:
I don't know if it's possible, but it might be cool to have a property that contains urls links to built data docs sites for this run. That would be helpful for building richer notifications.

Reply:
get_docs_sites_urls is valid for this too